### PR TITLE
add staking key/cert generation

### PR DESCRIPTION
**Pull request by danlaine**
* Remove cert generation scripts and artifacts
* Automatically create staking key/cert at `~/.gecko/staking` if key don't exist
* Rename `gecko/keys` to `gecko/staking` for clarity
* Improve CLI error handling
* Can use `~` in `--db-dir` and staking cert/key file paths

### CODE REVIEW

**Comment by moreati** [comment]
Would you consider instead of `~/.gecko`, a path that follows [XDG Base directory specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)? So the (default) path would be something like

| OS | Path |
| -- | -- |
| Linux/BSD | `~/.config/ava-labs/gecko/staking` |
| MacOS | `~/Library/Application Support/ava-labs/gecko/staking` |

https://github.com/OpenPeeDeeP/xdg looks like a reasonable implementation for Go.

**Comment by danlaine** [comment]
^ Thoughts on this, @StephenButtolph ?

**Comment by adasari** [comment]
+1 for `~/.gecko/staking`

File: main/params.go [hunk 1]
```diff
@@ -34,8 +35,14 @@ const (
 
 // Results of parsing the CLI
 var (
-	Config = node.Config{}
-	Err    error
+	Config                 = node.Config{}
+	Err                    error
+	defaultStakingKeyPath  = "~/.gecko/staking/staker.key"
```
[the remark this answers is not in the document]
Comment (danlaine) [reply main/params.go]: Good points...updated

**Comment by holisticode** [comment]
Intuitively I was associating XDG to *nix systems, but in fact I can also see that it is supported on Windows now, so I am liking this suggestion more and more as well :)

**Comment by danlaine** [comment]
@StephenButtolph addressed comments
Will leave XDG path for another issue/PR

**Comment by moreati** [comment]
I withdraw my suggestion on this PR. Replaced by a stand-alone enhancement proposal.

**Pull request merged**